### ISSUE TITLE
Why use layer normalization instead of batch normalization for wgan-gp model?

### ISSUE THREAD

**Issue by ybsave** [LynnHo/DCGAN-LSGAN-WGAN-WGAN-GP-Tensorflow#3]
It seems that the only difference between a regular discriminator model and a wgan-gp discriminator model is using batch norm or layer norm.

Why use layer norm in wgan-gp model? It seems that public believe that layer normalization does not perform as well as batch normalization for conv layers. 

Thank you.

**Comment by LynnHo** [comment]
@ybsave Because the gradient penalty should be independently imposed on different samples, but batch normalization brings batch correlation. It has been discussed in the paper of WGAN-GP.

**Comment by mrgransky** [comment]
From WGAN-GP [paper](https://arxiv.org/abs/1704.00028v3):

![bild](https://github.com/LynnHo/DCGAN-LSGAN-WGAN-GP-DRAGAN-Tensorflow-2/assets/11946010/3f7e9df6-9f17-4511-a51c-d5e3e482f20b)
